socorro/external/postgresql/new_crash_source.py

```diff
@@ -0,0 +1,242 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from datetime import timedelta
+from collections import Sequence
+
+from configman import Namespace, RequiredConfig, class_converter
+
+from socorro.lib.converters import (
+    change_default,
+    two_level_nested_list_converter
+)
+from socorro.lib.datetimeutil import (
+    utc_now,
+    string_to_datetime,
+)
+from socorro.external.postgresql.dbapi2_util import execute_query_fetchall
+
+
+#==============================================================================
+class PGQueryNewCrashSource(RequiredConfig):
+    """This class is an iterator that will yield a stream of crash_ids based
+    on a query to the PG database."""
+    required_config = Namespace()
+    required_config.add_option(
+        'transaction_executor_class',
+        default="socorro.database.transaction_executor."
+        "TransactionExecutorWithInfiniteBackoff",
+        doc='a class that will manage transactions',
+        from_string_converter=class_converter,
+        reference_value_from='resource.postgresql',
+    )
+    required_config.add_option(
+        'database_class',
+        default='socorro.external.postgresql.connection_context'
+            '.ConnectionContext',
+        doc='the class responsible for connecting to Postgres',
+        from_string_converter=class_converter,
+        reference_value_from='resource.postgresql',
+    )
+    required_config.add_option(
+        'crash_id_query',
+        doc='sql to get a list of crash_ids',
+        default="select 'some_id'",
+        likely_to_be_changed=True,
```

Comment:
let's say RelEng tagged a release incorrectly with the wrong channel, and we need to just those crashes to a special processor that could correct the problem, how would you do it?  By having the query readily available on the command line, I can reuse this class with a crashmover or submitter route all those crashes into wherever they need to go in minutes.  

If my only option was writing code, then a 30 minute job would become a 3 day ordeal of writing code, making a PR, getting review, sending to staging, making a release.